easybuild/easyconfigs/m/MATLAB-Engine/MATLAB-Engine-2021a-GCCcore-10.2.0.eb

```diff
@@ -0,0 +1,34 @@
+easyblock = 'PythonPackage'
+
+name = 'MATLAB-Engine'
+version = '2021a'
+
+homepage = 'https://www.mathworks.com/help/matlab/matlab-engine-for-python.html'
+description = """The MATLAB Engine API for Python provides a package for Python
+ to call MATLAB as a computational engine."""
+
+toolchain = {'name': 'GCCcore', 'version': '10.2.0'}
+
+dependencies = [
+    ('MATLAB', version, '', SYSTEM),
+    ('Python', '3.8.6'),
+]
+
+use_pip = False
+download_dep_fail = True
+
+# Since this setup.py does not provide a separate --build-base for the install
+# step, both build and install must be performed in a single command.
+prebuildopts = "cd $EBROOTMATLAB/extern/engines/python && "
+buildopts = "--build-base=%(builddir)s install --prefix=%(installdir)s"
+
+skipsteps = ['install']
+
+# Test that connection with MATLAB can be established successfully
+sanity_check_commands = ["python -c 'import matlab.engine; eng = matlab.engine.start_matlab(); eng.quit()'"]
+
+sanity_pip_check = True
+
+options = {'modulename': 'matlab.engine'}
+
```

Comment:
Loading the `matlabengine*.so` fails in systems with old glibc. We have to load the same glibc used by MATLAB:
```suggestion

# load glibc used by MATLAB binaries
modluafooter = """
    setenv("LD_PRELOAD", pathJoin(os.getenv("EBROOTMATLAB"), "bin", "glnxa64", "glibc-2.17_shim.so"))
"""

```

Reply:
this is also needed for version 2021b

Reply:
This will break things for using this in conjunction with other stuff from the module tree for everyone else though, since this is older than what others might have. It breaks matlab.engine itself for me on Rocky 8.5

I think this needs to be handled as a hook.

Reply:
you're right, since matlab works fine in our centos7 system I thought that it was always preloading that shim library, but it is not the case. It checks for the version of glibc and only preloads the shim library on systems with that specific version.
```
# Preload glibc_shim in case of RHEL7 variants
  test -e /usr/bin/ldd &&  ldd --version |  grep -q "(GNU libc) 2\.17"  \
            && export LD_PRELOAD="${MCRROOT}/bin/glnxa64/glibc-2.17_shim.so"
```

So I changed this suggestion to just a comment in https://github.com/Micket/easybuild-easyconfigs/pull/13